internal/cmd/base/format.go

```diff
@@ -134,58 +134,132 @@ func WrapMap(prefixSpaces, maxLengthOverride int, input map[string]interface{})
+			"Message": in.Message,
+		}
 
-	maxLength := MaxAttributesLength(nonAttributeMap, nil, nil)
+		if in.Op != "" {
+			nonAttributeMap["Operation"] = in.Op
+		}
 
-	ret := []string{
-		"",
-		"Error information:",
-		WrapMap(2, maxLength+2, nonAttributeMap),
-	}
+		maxLength := MaxAttributesLength(nonAttributeMap, nil, nil)
 
-	if in.Details != nil {
-		if len(in.Details.WrappedErrors) > 0 {
-			ret = append(ret,
-				"",
-				"  Wrapped Errors:",
-			)
-			for _, we := range in.Details.WrappedErrors {
-				ret = append(ret,
-					fmt.Sprintf("    Message:             %s", we.Message),
-					fmt.Sprintf("    Operation:           %s", we.Op),
-				)
-			}
+		var output []string
+		if contextStr != "" {
+			output = append(output, contextStr)
 		}
+		output = append(output,
+			"",
+			"Error information:",
+			WrapMap(2, maxLength+2, nonAttributeMap),
+		)
 
-		if len(in.Details.RequestFields) > 0 {
-			ret = append(ret,
-				"",
-				"  Field-specific Errors:",
-			)
-			for _, field := range in.Details.RequestFields {
-				if field.Name == "update_mask" {
-					// TODO: Report useful error messages related to "update_mask".
-					continue
+		if in.Details != nil {
+			if len(in.Details.WrappedErrors) > 0 {
+				output = append(output,
+					"",
+					"  Wrapped Errors:",
+				)
+				for _, we := range in.Details.WrappedErrors {
+					output = append(output,
+						fmt.Sprintf("    Message:             %s", we.Message),
+						fmt.Sprintf("    Operation:           %s", we.Op),
+					)
 				}
-				ret = append(ret,
-					fmt.Sprintf("    Name:              -%s", strings.ReplaceAll(field.Name, "_", "-")),
-					fmt.Sprintf("      Error:           %s", field.Description),
+			}
+
+			if len(in.Details.RequestFields) > 0 {
+				output = append(output,
+					"",
+					"  Field-specific Errors:",
 				)
+				for _, field := range in.Details.RequestFields {
+					if field.Name == "update_mask" {
+						// TODO: Report useful error messages related to "update_mask".
+						continue
+					}
+					output = append(output,
+						fmt.Sprintf("    Name:              -%s", strings.ReplaceAll(field.Name, "_", "-")),
+						fmt.Sprintf("      Error:           %s", field.Description),
+					)
+				}
 			}
 		}
+
+		c.UI.Output(WrapForHelpText(output))
+	}
+}
+
+// PrintCliError prints the given CLI error to the UI in the appropriate format
+func (c *Command) PrintCliError(err error) {
+	switch Format(c.UI) {
+	case "table":
+		c.UI.Error(err.Error())
+	case "json":
+		output := struct {
+			Error string `json:"error"`
+		}{
+			Error: err.Error(),
+		}
+		b, _ := JsonFormatter{}.Format(output)
+		c.UI.Output(string(b))
+	}
+}
+
+// PrintJsonItem prints the given item to the UI in JSON format
+func (c *Command) PrintJsonItem(result api.GenericResult, item interface{}) int {
+	output := struct {
+		StatusCode int         `json:"status_code"`
+		Item       interface{} `json:"item"`
+	}{
+		StatusCode: result.GetResponse().HttpResponse().StatusCode,
```

Comment:
Rather than adding it to GenericResult I added it to `api.Response` as it's really a property of the response, specifically the HTTP response. But there it seems to make sense to pull it up so you don't do `Response().HttpResponse().StatusCode`.